components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AbstractOTPAuthenticator.java

```diff
@@ -0,0 +1,1243 @@
+import static org.wso2.carbon.identity.application.authentication.framework.util.AbstractOTPAuthenticatorConstants.IDF_HANDLER_NAME;
+import static org.wso2.carbon.identity.application.authentication.framework.util.AbstractOTPAuthenticatorConstants.INVALID_USERNAME;
+import static org.wso2.carbon.identity.application.authentication.framework.util.AbstractOTPAuthenticatorConstants.IS_LOGIN_ATTEMPT_BY_INVALID_USER;
+import static org.wso2.carbon.identity.application.authentication.framework.util.AbstractOTPAuthenticatorConstants.LOCAL_AUTHENTICATOR;
+import static org.wso2.carbon.identity.application.authentication.framework.util.AbstractOTPAuthenticatorConstants.OTP;
+import static org.wso2.carbon.identity.application.authentication.framework.util.AbstractOTPAuthenticatorConstants.OTP_ALPHA_NUMERIC_CHAR_SET;
+import static org.wso2.carbon.identity.application.authentication.framework.util.AbstractOTPAuthenticatorConstants.OTP_NUMERIC_CHAR_SET;
+import static org.wso2.carbon.identity.application.authentication.framework.util.AbstractOTPAuthenticatorConstants.OTP_RESEND_ATTEMPTS;
+import static org.wso2.carbon.identity.application.authentication.framework.util.AbstractOTPAuthenticatorConstants.RECAPTCHA_PARAM;
+import static org.wso2.carbon.identity.application.authentication.framework.util.AbstractOTPAuthenticatorConstants.RESEND;
+import static org.wso2.carbon.identity.application.authentication.framework.util.AbstractOTPAuthenticatorConstants.RETRY_QUERY_PARAMS;
+import static org.wso2.carbon.identity.application.authentication.framework.util.AbstractOTPAuthenticatorConstants.SCREEN_VALUE_QUERY_PARAM;
+import static org.wso2.carbon.identity.application.authentication.framework.util.AbstractOTPAuthenticatorConstants.UNLOCK_QUERY_PARAM;
+import static org.wso2.carbon.identity.application.authentication.framework.util.AbstractOTPAuthenticatorConstants.USERNAME;
+import static org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants.RequestParams.IS_IDF_INITIATED_FROM_AUTHENTICATOR;
+import static org.wso2.carbon.identity.event.IdentityEventConstants.Event.POST_NON_BASIC_AUTHENTICATION;
+import static org.wso2.carbon.identity.event.IdentityEventConstants.EventProperty.AUTHENTICATOR_NAME;
+import static org.wso2.carbon.identity.event.IdentityEventConstants.EventProperty.OPERATION_STATUS;
+import static org.wso2.carbon.identity.event.IdentityEventConstants.EventProperty.PROPERTY_FAILED_LOGIN_ATTEMPTS_CLAIM;
+import static org.wso2.carbon.identity.event.IdentityEventConstants.EventProperty.USER_STORE_MANAGER;
+import static org.wso2.carbon.user.core.UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME;
+
+/**
+ * This class contains the implementation of abstract OTP authenticator.
+ */
+public abstract class AbstractOTPAuthenticator extends AbstractApplicationAuthenticator {
+
+    private static final Log log = LogFactory.getLog(AbstractOTPAuthenticator.class);
```

Comment:
Since this is a constant, make sure that the constant name is in the correct format.